hack/debug.sh

```diff
@@ -73,7 +90,13 @@ do_master () {
     mkdir -p $logmaster
 
     # Log the generic system stuff
-    log_system $logmaster $aos_master_service
+    services=
+    for service in "${services[@]}"; do
+        if [ -n "${service}" ]; then
+            services="${services} ${service}"
+        fi
```

Comment:
You're overwriting the earlier "services" variable (`services=($aos_master_service $aos_master_controllers_service $aos_master_api_service)`) here so this loop is going to just fail isn't it? If it does actually work due to perl-like separation of scalar and array variables, it's at least incredibly confusing...